src/middleware.js

```diff
@@ -38,14 +38,14 @@ function _getContentType(headers) {
 }
 
 function _validateRequest(requestOptions) {
-    return Promise.all([
-        _validateParams(requestOptions.headers, requestOptions.params, requestOptions.query, requestOptions.files, requestOptions.path, requestOptions.method.toLowerCase()).catch(e => e),
-        _validateBody(requestOptions.body, requestOptions.path, requestOptions.method.toLowerCase(), _getContentType(requestOptions.headers)).catch(e => e)
-    ]).then(function (errors) {
-        if (errors[0] || errors[1]) {
-            return errors[0] && errors[1] ? Promise.reject(errors[0].concat(errors[1])) : errors[0] ? Promise.reject(errors[0]) : Promise.reject(errors[1]);
-        }
-    }).catch(function (errors) {
+    const paramValidationErrors = _validateParams(requestOptions.headers, requestOptions.params, requestOptions.query, requestOptions.files, requestOptions.path, requestOptions.method.toLowerCase());
```

Comment:
what do you say about using here `|| []` or return `[]` instead of `undefined` from `_validateParams` and `_validateBody`?

in this case we could just do in line 44
```js
const errors = paramValidationErrors.concat[bodyValidationErrors]
```

Reply:
i'm not sure we don't just pass and destructure `requestOptions` in `_validateParams` and `_validateBody` instead. what do you say about changing this unreadable piece of code? :)

Reply:
IMO, changing `_validateRequest` to be a sync function should be also documented as breaking change in the changelog.

Reply:
@kobik Done.
Why would internal change that is in no way exposed to the outside world be a breaking change? The way middleware behaves has not changed.